Reserved DriveLetter in Unlock-Failed-Dialog not displayed

### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/cryptomator/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/cryptomator/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

The unlock failed dialog does not show the correct drive letter.

### What software is involved?

- Operating System: Windows 10 Build
- Cryptomator: 1.6.10


### Volume Type

_No response_

### Steps to Reproduce

1. Create two vaults (A and B)
2. For each, set in vault options mountpoint to the _same_ fixed drive letter (e.g. `Y:\`)
3. Unlock A
4. Unlock B


### Expected Behavior

Unlock failed dialog for B is showing that drive letter Y is reserved.

### Actual Behavior

Unlock failed dialog for B is showing that drive letter "AUTO" is reserved.

### Reproducibility

Always

### Relevant Log Output

_No response_

### Anything else?

Screenshot:
![grafik](https://user-images.githubusercontent.com/9036915/176130649-9d4549a2-06a4-4589-b0d7-ea82a312c0f5.png)

The situation seem to have gotten worse since 6395f177363ddafe85e2d02e6d1795eedd48e015 which causes now a stacktrace to be shown instead of an error message, see

```
Error Code 318Q:206H:OVHF
org.cryptomator.integrations.mount.MountFailedException: org.cryptomator.jfuse.api.FuseMountFailedException: fuse_mount failed
	at org.cryptomator.frontend.fuse@3.0.0/org.cryptomator.frontend.fuse.mount.WinFspMountProvider$WinFspMountBuilder.mount(WinFspMountProvider.java:128)
	at org.cryptomator.desktop@1.9.1/org.cryptomator.common.mount.Mounter.mount(Mounter.java:134)
	at org.cryptomator.desktop@1.9.1/org.cryptomator.common.vaults.Vault.unlock(Vault.java:150)
	at org.cryptomator.desktop@1.9.1/org.cryptomator.ui.keyloading.KeyLoadingStrategy.use(KeyLoadingStrategy.java:79)
	at org.cryptomator.desktop@1.9.1/org.cryptomator.ui.keyloading.KeyLoadingStrategy.use(KeyLoadingStrategy.java:84)
	at org.cryptomator.desktop@1.9.1/org.cryptomator.ui.unlock.UnlockWorkflow.attemptUnlock(UnlockWorkflow.java:70)
	at org.cryptomator.desktop@1.9.1/org.cryptomator.ui.unlock.UnlockWorkflow.call(UnlockWorkflow.java:60)
	at org.cryptomator.desktop@1.9.1/org.cryptomator.ui.unlock.UnlockWorkflow.call(UnlockWorkflow.java:31)
	at javafx.graphics@20.0.1/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.cryptomator.jfuse.api.FuseMountFailedException: fuse_mount failed
	at org.cryptomator.jfuse.win@0.5.1/org.cryptomator.jfuse.win.FuseImpl.mount(FuseImpl.java:48)
	at org.cryptomator.jfuse.api@0.5.1/org.cryptomator.jfuse.api.Fuse.mount(Fuse.java:127)
	at org.cryptomator.jfuse.win@0.5.1/org.cryptomator.jfuse.win.FuseImpl.mount(FuseImpl.java:37)
	at org.cryptomator.frontend.fuse@3.0.0/org.cryptomator.frontend.fuse.mount.WinFspMountProvider$WinFspMountBuilder.mount(WinFspMountProvider.java:125)
	... 14 more

```

I'm trying to address that in https://github.com/cryptomator/cryptomator/pull/2961.

--- a/src/main/java/org/cryptomator/common/mount/MountPointInUseException.java
+++ b/src/main/java/org/cryptomator/common/mount/MountPointInUseException.java
@@ -1,0 +1,8 @@
+package org.cryptomator.common.mount;
+
+public class MountPointInUseException extends IllegalMountPointException {
+
+	public MountPointInUseException(String msg) {
+		super(msg);
+	}
+}

--- a/src/main/java/org/cryptomator/common/mount/Mounter.java
+++ b/src/main/java/org/cryptomator/common/mount/Mounter.java
@@ -97,7 +97,11 @@ public class Mounter {
 				}
 			} else {
 				var mpIsDriveLetter = userChosenMountPoint.toString().matches("[A-Z]:\\\\");
-				if (!mpIsDriveLetter && canMountToParent && !canMountToDir) {
+				if (mpIsDriveLetter) {
+					if (driveLetters.getOccupied().contains(userChosenMountPoint)) {
+						throw new MountPointInUseException(userChosenMountPoint.toString());
+					}
+				} else if (canMountToParent && !canMountToDir) {
 					MountWithinParentUtil.prepareParentNoMountPoint(userChosenMountPoint);
 					cleanup = () -> {
 						MountWithinParentUtil.cleanup(userChosenMountPoint);

--- a/src/main/java/org/cryptomator/ui/unlock/UnlockInvalidMountPointController.java
+++ b/src/main/java/org/cryptomator/ui/unlock/UnlockInvalidMountPointController.java
@@ -1,5 +1,6 @@
 package org.cryptomator.ui.unlock;
 
+import org.cryptomator.common.mount.MountPointInUseException;
 import org.cryptomator.common.mount.MountPointNotExistsException;
 import org.cryptomator.common.mount.MountPointNotSupportedException;
 import org.cryptomator.common.vaults.Vault;
@@ -41,6 +42,7 @@ public class UnlockInvalidMountPointController implements FxController {
 		var translationKey = switch (e) {
 			case MountPointNotSupportedException x -> "unlock.error.customPath.description.notSupported";
 			case MountPointNotExistsException x -> "unlock.error.customPath.description.notExists";
+			case MountPointInUseException x -> "unlock.error.customPath.description.inUse";
 			default -> "unlock.error.customPath.description.generic";
 		};
 		dialogDescription.setFormat(resourceBundle.getString(translationKey));

--- a/src/main/resources/i18n/strings.properties
+++ b/src/main/resources/i18n/strings.properties
@@ -127,6 +127,7 @@ unlock.success.revealBtn=Reveal Drive
 unlock.error.customPath.message=Unable to mount vault to custom path
 unlock.error.customPath.description.notSupported=If you wish to keep using the custom path, please go to the preferences and select a volume type that supports it. Otherwise, go to the vault options and choose a supported mount point.
 unlock.error.customPath.description.notExists=The custom mount path does not exist. Either create it in your local filesystem or change it in the vault options.
+unlock.error.customPath.description.inUse=Drive letter "%s" is already in use.
 unlock.error.customPath.description.generic=You have selected a custom mount path for this vault, but using it failed with the message: %s
 ## Hub
 hub.noKeychain.message=Unable to access device key

--- a/suppression.xml
+++ b/suppression.xml
@@ -55,4 +55,12 @@
 		<cve>CVE-2022-45688</cve>
 	</suppress>
 
+	<suppress>
+		<notes><![CDATA[
+		False positive for jackson-databind-2.14.2.jar, see https://github.com/FasterXML/jackson-databind/issues/3972
+   ]]></notes>
+		<packageUrl regex="true">^pkg:maven/com\.fasterxml\.jackson\.core/jackson\-databind@.*$</packageUrl>
+		<cve>CVE-2023-35116</cve>
+	</suppress>
+
 </suppressions>

